VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/LoopedRandomInstantiation/LoopedRandomInstantiationAnalyzer.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using VSDiagnostics.Utilities;
+
+namespace VSDiagnostics.Diagnostics.General.LoopedRandomInstantiation
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class LoopedRandomInstantiationAnalyzer : DiagnosticAnalyzer
+    {
+        private const DiagnosticSeverity Severity = DiagnosticSeverity.Warning;
+        private static readonly string Category = VSDiagnosticsResources.GeneralCategory;
+        private static readonly string Message = VSDiagnosticsResources.LoopedRandomInstantiationAnalyzerMessage;
+        private static readonly string Title = VSDiagnosticsResources.LoopedRandomInstantiationAnalyzerTitle;
+
+        internal static DiagnosticDescriptor Rule =>
+                new DiagnosticDescriptor(DiagnosticId.LoopedRandomInstantiation, Title, Message, Category, Severity, true);
+
+        private SyntaxKind[] _loopTypes = {SyntaxKind.ForEachStatement, SyntaxKind.ForStatement, SyntaxKind.WhileStatement};
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(AnalyzeSymbol, SyntaxKind.VariableDeclaration);
+        }
+
+        private void AnalyzeSymbol(SyntaxNodeAnalysisContext context)
+        {
+            var variableDeclaration = context.Node as VariableDeclarationSyntax;
+
+            var type = variableDeclaration?.Type;
+            if (type == null) { return; }
+
+            var typeInfo = context.SemanticModel.GetTypeInfo(type).Type;
+            if (typeInfo == null) { return; }
+
+            if (typeInfo.OriginalDefinition.ContainingNamespace.Name != nameof(System) ||
+                typeInfo.Name != nameof(System.Random))
+            {
+                return;
+            }
+
+            SyntaxNode currentNode = variableDeclaration;
+            while (!currentNode.IsKind(SyntaxKind.ClassDeclaration))
+            {
+                if (_loopTypes.Contains(currentNode.Kind()))
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Rule, variableDeclaration.GetLocation()));
```

Comment:
Can you create a separate warning for every `VariableDeclarator`? That way we have 2 warnings for for example `var r = new Random(), x = new Random();`.